KubeArmor/core/kubeArmor.go

```diff
@@ -577,8 +614,16 @@ func KubeArmor() {
 		dm.Logger.Print("Started to monitor host security policies")
 	}
 
+	policyService := &policy.ServiceServer{}
+
+	if !dm.K8sEnabled && cfg.GlobalCfg.Policy {
+		if _, err := os.Stat(trimmedSocket); err == nil {
+			policyService.UpdateContainerPolicy = dm.ParseAndUpdateContainerSecurityPolicy
+			dm.Logger.Print("Started to monitor container security policies on gRPC")
+		}
+	}
```

Comment:
```suggestion
	if enableContainerPolicy {
		policyService.UpdateContainerPolicy = dm.ParseAndUpdateContainerSecurityPolicy
		dm.Logger.Print("Started to monitor container security policies on gRPC")
	}
```

Reply:
Don't we need this check? Added it for the condition when we don't set the criSocket flag and the cri socket file is also not available on the system.